File: packages/api/cmd/sikuligrpc/examples/node/ocr.mjs
import { ensureSikuligoOnPath } from "./bootstrap.mjs";
import { Sikuli } from "@sikuligo/sikuligo";

ensureSikuligoOnPath();

function grayImageFromRows(name, rows) {
  const height = rows.length;
  const width = rows[0].length;
  const pix = rows.flat().map((v) => v & 0xff);
  return {
    name,
    width,
    height,
    pix: Buffer.from(pix)
  };
}

const client = await Sikuli();
const source = grayImageFromRows("ocr-source", [
  [220, 220, 220, 220],
  [220, 20, 20, 220],
  [220, 220, 220, 220]
]);

try {
  const readText = await client.readText({
    source,
    params: {
      language: "eng"
    }
  });
  console.log("readText", JSON.stringify(readText, null, 2));

  const findText = await client.findText({
    source,
    query: "example",
    params: {
      language: "eng",
      case_sensitive: false
    }
  });
  console.log("findText", JSON.stringify(findText, null, 2));
} finally {
  await client.close();
}

File: packages/client-python/examples/ocr.py
from __future__ import annotations

from bootstrap import ensure_sikuligo_on_path
from generated.sikuli.v1 import sikuli_pb2 as pb
from sikuligo import Screen
from sikuligo.client import gray_image_from_rows

ensure_sikuligo_on_path()

screen = Screen()
client = screen.client
try:
    source = gray_image_from_rows(
        "ocr-source",
        [
            [220, 220, 220, 220],
            [220, 20, 20, 220],
            [220, 220, 220, 220],
        ],
    )

    read_req = pb.ReadTextRequest(source=source, params=pb.OCRParams(language="eng"))
    read_res = client.read_text(read_req, timeout_seconds=10.0)
    print(f"read_text => {read_res.text!r}")

    find_req = pb.FindTextRequest(
        source=source,
        query="example",
        params=pb.OCRParams(language="eng", case_sensitive=False),
    )
    find_res = client.find_text(find_req, timeout_seconds=10.0)
    print(f"find_text matches => {len(find_res.matches)}")
finally:
    screen.close()
